### PR TITLE
Feature flow control grammar

### PR DESCRIPTION
**Pull request by phddoom**
**Problem**: Address #119 and #158

**Solution**:  Push `if` `else` `for` `fn` `end` statements into the grammar / parser.

**Changes introduced by this pull request**:

- Expanded `Statement` Enum
- `Comparitor` Enum for handling `if` statement comparisons
- Added tests for parsing new statements

**Drawbacks**: I am not fluent in Rust yet so the code might not be idiomatic Rust.

**TODOs**:

- [x] Remove commented code
- [x] Remove dead code
- [ ] Refactor for more Rustic feel

**Fixes**: #119 and #158

**State**: WIP

**Blocking/related**: None currently known

**Other**:

------

### CODE REVIEW

**Comment by skylerberg** [comment]
This is a really important and challenging change. Thanks for tackling it. I haven't had time for the project recently, so I don't have the time to give a thorough review. You marked this as work in progress; do you want me to hold off on merging this?

**Comment by phddoom** [comment]
@skylerberg Yes please hold off on merging this. I would like to add some example tests and clean up the code. Would it be best to mention you again when this PR is ready to merge?

**Comment by skylerberg** [comment]
@phddoom Yes, that would be perfect.

**Comment by phddoom** [comment]
@skylerberg I think this is ready to merge now. Would you like me to squash my commits? And rewrite the commit mesage before merging?

File: src/grammar.rustpeg [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+comment -> ()
+    = [#] [^\r\n]*
+
+whitespace -> ()
+    = [ \t]+
+
+job_ending -> ()
+    = [;]
+    / newline
+    / newline
```
Comment (skylerberg) [inline src/grammar.rustpeg]: Looks like this got duplicated.

**Comment by skylerberg** [comment]
No need to squash, your commits have well formatted and descriptive messages representing mostly independent steps.

In this PR you are tackling one of the most important changes needed in Ion IMO. Your code has tests and you took time to clean everything up. In short, you are awesome! Thanks a ton for this contribution.

**Pull request merged**